drafts/inlinemacros.md

```diff
@@ -165,99 +170,66 @@ positions:
 Inline types can specifically not be used as
 
  - types of abstract methods or fields
- - types of variables
- - types of class parameters
+ - types of vars
+ - types of parameters to non-inline classes
  - type parameters to methods
  - type parameters in `new` expressions or class parents.
 
-## Inline Overloading Resolution
-
-We have stated that the rewriting is done on typed trees and that references of
-the original tree are kept after the rewrite. There is one exception to this: References to
-overloaded methods are re-computed after the rewrite. So a different alternative might
-be chosen if the argument types after the rewrite are more specific than the method
-parameter types. So, compile-time overloading resolution is essentially multi-method dispatch.
-Here is a scenario where this can be used to support constant folding inline infix methods.
-
-The task here is to define `++` as an infix operator for inline Ints, which should
-give an inline result if and only if its two operands are inline values. Here's how it can be achieved:
-
-    inline implicit class PlusWrapper(val x: Int) {
-      inline def ++ (y: Int): Int = plus(this.x, y)
-    }
-    inline def plus(x: inline Int, y: inline Int): inline Int = x + y
-    def plus(x: Int, y: Int): Int = x + y
-
-Let's assume we have
-
-    a, b: inline Int
-
-Then we rewrite `a ++ b` as follows:
-
-    a ++ b
-    -->
-    PlusWrapper(a).++(b)
-    -->
-    new PlusWrapper(a).++(b)
-    -->
-    plus(new PlusWrapper(a).x, b)
-    -->
-    plus(a, b)
-    -->
-    a + b: inline Int
-
-Note that the call to `plus` in `PlusWrapper` resolves statically to the non-inline method `plus`.
-Yet in the last-but-one expression of the above rewrite sequence it is rewritten to refer to the
-inline method `plus` because at that point both arguments are known to be inline values.
-
-
 ## Macros
 
-A macro is an expression of the form
+A macro expression is an expression of the form
```

Comment:
As agreed upon in one of the emails, changed the terminology to say `macro expression` instead of `macro`. Not only here, but everywhere in the document.